.github/workflows/ci.yml

```diff
@@ -12,98 +12,144 @@ concurrency:
 # ${{ vars.CI_UNIFIED_IMAGE }} is defined in the repository variables
 
 jobs:
-  check:
-    name: "check"
+  fmt:
+    name: Cargo fmt
     runs-on: ubuntu-latest
     container:
       image: ${{ vars.CI_IMAGE }}
     steps:
-      - uses: actions/checkout@v3
+      - name: Checkout sources
+        uses: actions/checkout@v4
 
-      # uncomment if needed
-      # - name: fmt
-      #   run: |
-      #     cargo fmt --all --check
+      - name: Install Rust nightly toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+          components: rustfmt
 
       - name: Rust Cache
-        uses: Swatinem/rust-cache@v2.5.0
+        uses: Swatinem/rust-cache@23bce251a8cd2ffc3c1075eaa2367cf899916d84 # v2.7.3
         with:
           cache-on-failure: true
           cache-all-crates: true
 
-      - name: check
-        run: cargo check
+      - name: Cargo fmt
+        run: cargo +nightly fmt --all -- --check
 
-  test:
-    name: "test"
+      - if: "failure()"
+        uses: "andymckay/cancel-action@271cfbfa11ca9222f7be99a47e8f929574549e0a" # v0.4
```

Comment:
Does this cancel `Cargo fmt` check only, or all the CI checks? I would make checks to be independent.

Reply:
This would cancel all running CI steps. We've been using it in subxt to reduce the CI minutes in case of failure.
However, I've removed this from litep2p since I believe we don't consume that many minutes 👍